agent/http.go

```diff
@@ -949,25 +965,33 @@ func (s *HTTPServer) parseTokenInternal(req *http.Request, token *string) {
 
 			// <Scheme> must be "Bearer"
 			if strings.ToLower(scheme) == "bearer" {
-				// Since Bearer tokens shouldnt contain spaces (rfc6750#section-2.1)
+				// Since Bearer tokens shouldn't contain spaces (rfc6750#section-2.1)
 				// "value" is tokenized, only the first item is used
 				tok = strings.TrimSpace(strings.Split(value, " ")[0])
 			}
 		}
 	}
 
-	if tok != "" {
-		*token = tok
+	*token = tok
+	return
+}
+
+// parseTokenResolveProxy passes through to parseTokenInternal and optionally resolves proxy tokens to real ACL tokens.
```

Comment:
We don't have "Proxy" tokens anymore. That was something we did away with when we got rid of "managed" connect proxies (Consul would spawn a subprocess to run the proxy and manage its life cycle. It would provide it some agent specific token that could then be translated back to a real ACL token).

So maybe a better name for this function would be `parseTokenOrDefault` as that is basically all its doing?

Reply:
Sounds good, updating the name!